src/routes/sunshine.js

```diff
@@ -94,8 +95,54 @@ module.exports = (app, db) => {
         res.status(200).json(db.getFields());
     };
 
+    const exportHandler = async (req, res) => {
+        const query = queryFromReq(req);
+
+        // headers - array of strings
+        const headersToCSV = (headers) => {
+            var line = '';
+            for (header of headers) {
+                line += '"' + header.toString() + '",';
+            }
+            line += '\n';
+            return line;
+        }
+
+        // entry - an object from database to convert to a CSV line
+        // headers - array of headers
+        const entryToCSV = (entry, headers) => {
```

Comment:
Same as above, this function could be reduced quite a bit using array functions:
```JavaScript
const entryToCSV = (entry, headers) => headers
    .map((header) => entry[header] ? `"${entry[header].toString()}"` : '""')
    .join(',') + '\n';
```